### PR TITLE
Add modal dialogs

### PR DESCRIPTION
**Pull request by yscik**
Part of 186-gh-Automattic/wpjm-addons

Includes some related CSS fixes that came up along the way.

### Changes Proposed in this Pull Request

* Add a new modal dialog class that can open a popup modal in an overlay
* Uses the native `<dialog>` element
* Add styling and scripts to open and close the dialog
* Move styles and code around for some elements (buttons), to share it between notice, dialog and other future classes
* Fix the buttons and links not picking up theme colors
* Fix alignment issue in hint notices (438-gh-Automattic/wpjm-addons#issuecomment-1892794397), and add a new variation for 436-gh-Automattic/wpjm-addons#discussion_r1453638803

### Testing Instructions

* Add the `[job_manager_ui_test]` shortcode to a page, publish it and open it
* First element should have an 'Open Modal' button, use it to open the modal dialog
* Clicking outside, on the X or the Close button should close the modal
* Double-check that the rest of the notices keep working

### Release Notes

* Add support for modal dialogs

### Screenshot / Video

Desktop:

image

Mobile:

image

----

| Plugin build for ab543ddaa4f796fbe2019c38a2dc35dada63ac1e <a href="#"></a> |
| ------------------------------------------------------------ |
| 📦 [Download plugin zip](https://wpjobmanager.com/wp-content/uploads/2024/01/wp-job-manager-zip-2706-ab543dda.zip)                       |
| ▶️ [Open in playground](https://wpjobmanager.com/playground/?core=2024/01/2706-ab543dda)             |

### CODE REVIEW

**Comment by yscik** [comment]
Fixed style loading for non-block themes in ead3c17, and rebased to latest trunk.

File: includes/ui/class-notice.php [hunk 1]
```diff
@@ -32,10 +32,10 @@ public static function success( $args ) {
 
 		return self::render(
 			array_merge(
+				$args,
```
Comment (pgk) [inline includes/ui/class-notice.php]: If these are flipped it means that passing an icon change in $args will not take precedence. Is this somethihg we want?

File: includes/ui/class-ui-elements.php [hunk 2]
```diff
@@ -0,0 +1,176 @@
+<?php
+/**
+ * Notice component for the frontend.
```
Comment (pgk) [inline includes/ui/class-ui-elements.php]: I think this "Notice" comment is not relevant to what the class does. It seems to be encapsulating various UI utils and elements.

**Review by pgk — APPROVED** [review]
Modal worked well, left some nitpicky comments, feel free to address as needed

File: includes/ui/class-notice.php (hunk 1 above)
Comment (yscik) [reply includes/ui/class-notice.php]: Good catch, thanks. Did this to fix the `classes` merges, tweaking it a bit more in ab543dd so the icon overwrites keep working too.

**Pull request merged**